internal/core/liquid.go

```diff
@@ -0,0 +1,471 @@
+/*******************************************************************************
+*
+* Copyright 2024 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package core
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"slices"
+
+	"github.com/gophercloud/gophercloud/v2"
+	. "github.com/majewsky/gg/option"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/sapcc/go-api-declarations/limes"
+	limesrates "github.com/sapcc/go-api-declarations/limes/rates"
+	"github.com/sapcc/go-api-declarations/liquid"
+	"github.com/sapcc/go-bits/liquidapi"
+	"github.com/sapcc/go-bits/logg"
+	"github.com/sapcc/go-bits/promquery"
+
+	"github.com/sapcc/limes/internal/db"
+)
+
+// LiquidConnection holds all the necessary information which is necessary to interact with the LiquidClient.
+// In future, the state information of the LiquidConnection should be persisted in the database to avoid
+// reloading in case of configuration changes.
+type LiquidConnection struct {
+	// configuration
+	Area                            string
+	LiquidServiceType               string
+	ServiceType                     db.ServiceType
+	FixedCapacityConfiguration      Option[FixedCapacityConfiguration]      `yaml:"fixed_capacity_values"`
+	PrometheusCapacityConfiguration Option[PrometheusCapacityConfiguration] `yaml:"capacity_values_from_prometheus"`
+
+	// state
+	LiquidServiceInfo liquid.ServiceInfo `yaml:"-"`
+	LiquidClient      LiquidClient       `yaml:"-"`
+}
+
+// MakeLiquidConnection is a factory to fill all necessary configuration fields
+func MakeLiquidConnection(lc LiquidConfiguration) LiquidConnection {
```

Comment:
By convention, factory functions are called `NewSomething` in Go.
```suggestion
// NewLiquidConnection is a factory to fill all necessary configuration fields
func NewLiquidConnection(lc LiquidConfiguration) LiquidConnection {
```

Reply:
I am very sureI read somewhere, that `MakeSomething` is used when returning a struct and `NewSomething` is used when returning a pointer. Is this a thing? I see the code has this nowhere, so I'll change it.